### PR TITLE
Adds default inserted_at to Testing.perform_job/3

### PR DESCRIPTION
**Pull request by antedeguemon**
Hey there! 🖖 

This pull request adds a default `inserted_at` to jobs executed by `Testing.perform_job/3`.

The objective is to make it behave more like a job executed by a regular pipeline - where `inserted_at` will always be populated.

Kudos to @amishpatel1994

### CODE REVIEW

File: test/oban/testing_test.exs [hunk 1]
```diff
@@ -74,8 +74,14 @@ defmodule Oban.TestingTest do
...
-    def perform(%{attempted_at: attempted_at, scheduled_at: scheduled_at}) do
-      {:ok, {attempted_at, scheduled_at}}
+    def perform(%Oban.Job{} = job) do
+      result = %{
+        attempted_at: job.attempted_at,
+        scheduled_at: job.scheduled_at,
+        inserted_at: job.inserted_at
+      }
+
+      {:ok, result}
```
Comment (sorentwo) [inline test/oban/testing_test.exs:84]:
```suggestion
      {:ok, Map.take(job, ~w(attempted_at inserted_at scheduled_at)a}
```

File: test/oban/testing_test.exs [hunk 2]
```diff
@@ -185,14 +191,33 @@ defmodule Oban.TestingTest do
...
 
-    test "retaining attempted_at or scheduled_at timestamps" do
-      time = ~U[2020-02-20 00:00:00.000000Z]
+    test "defaulting timestamp fields to mimic real execution" do
+      assert {:ok,
+              %{
+                attempted_at: %DateTime{},
+                scheduled_at: %DateTime{},
+                inserted_at: %DateTime{}
+              }} = perform_job(TemporalWorker, %{})
```
Comment (sorentwo) [inline test/oban/testing_test.exs:200]: Not quite as accurate, but what's important is that they're not nil:

```suggestion
      assert {:ok, result} = perform_job(TemporalWorker, %{})

      assert %{attempted_at: %_{}, inserted_at: %_{}, scheduled_at: %_{}} = result
```

File: test/oban/testing_test.exs [hunk 3]
```diff
@@ -185,14 +191,33 @@ defmodule Oban.TestingTest do
...
+              }} =
+               perform_job(TemporalWorker, %{},
+                 attempted_at: time,
+                 scheduled_at: time,
+                 inserted_at: time
+               )
+
+      assert attempted_at == time
+      assert scheduled_at == time
+      assert inserted_at == time
```
Comment (sorentwo) [inline test/oban/testing_test.exs:220]:
```suggestion
      assert {:ok, result} =
               perform_job(TemporalWorker, %{},
                 attempted_at: time,
                 inserted_at: time,
                 scheduled_at: time
               )

      assert result.attempted_at == time
      assert result.inserted_at == time
      assert result.scheduled_at == time
```

**Review by sorentwo — CHANGES_REQUESTED** [review]
Thank you for the fix! I neglected that timestamp when I updated that code quite recently. 

I've left a few nitpicks intended to avoid large blocks of vertical/hanging pattern matching because it's a pet peeve of mine.

File: test/oban/testing_test.exs (hunk 1 above)
Comment (antedeguemon) [reply test/oban/testing_test.exs]: @sorentwo it is missing the `)` 🙈 
```elixir
     {:ok, Map.take(job, ~w(attempted_at inserted_at scheduled_at)a)}
```

**Review by sorentwo — APPROVED** [review]

**Comment by sorentwo** [comment]
Thanks again 💛

**Pull request merged**